cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/RouteConfigHttpHandler.java

```diff
@@ -78,7 +78,14 @@ public void storeRouteConfig(HttpRequest request, HttpResponder responder,
                                @PathParam("service-id") String serviceId) throws Exception {
     ProgramId programId = Ids.namespace(namespaceId).app(appId).service(serviceId);
     Map<String, Integer> routes = parseBody(request, ROUTE_CONFIG_TYPE);
-    routeStore.store(programId, new RouteConfig(routes));
+    RouteConfig routeConfig;
+    try {
+      routeConfig = new RouteConfig(routes);
```

Comment:
BadRequestException is related to our Http handler logic. Why should RouteConfig constructor throw that exception?